src/darwin/Framework/CHIPTests/MTRPairingTests.m

```diff
@@ -227,23 +157,51 @@ + (void)shutdownStack
     [[MTRDeviceControllerFactory sharedInstance] stopControllerFactory];
 }
 
-- (void)test000_SetUp
+- (void)setUp
 {
-    // Nothing to do here; our setUp method handled this already.  This test
-    // just exists to make the setup not look like it's happening inside other
-    // tests.
+    [super setUp];
+    [self setContinueAfterFailure:NO];
+}
+
+// attestationDelegate and failSafeExtension can both be nil
+- (void)doPairingTestWithAttestationDelegate:(id<MTRDeviceAttestationDelegate>)attestationDelegate failSafeExtension:(NSNumber *)failSafeExtension
+{
+    // Don't reuse node ids, because that will confuse us.
+    ++sDeviceId;
+    XCTestExpectation * expectation = [self expectationWithDescription:@"Commissioning Complete"];
+
+    __auto_type * controllerDelegate = [[MTRPairingTestControllerDelegate alloc] initWithExpectation:expectation
+                                                                                 attestationDelegate:attestationDelegate
+                                                                                   failSafeExtension:failSafeExtension];
+    dispatch_queue_t callbackQueue = dispatch_queue_create("com.chip.pairing", DISPATCH_QUEUE_SERIAL_WITH_AUTORELEASE_POOL);
+
+    [sController setDeviceControllerDelegate:controllerDelegate queue:callbackQueue];
+
+    NSError * error;
+    __auto_type * payload = [MTRSetupPayload setupPayloadWithOnboardingPayload:kOnboardingPayload error:&error];
+    XCTAssertNotNil(payload);
+    XCTAssertNil(error);
```

Comment:
I know you just moved this, but
```suggestion
    XCTAssertNil(error);
    XCTAssertNotNil(payload);
```
gives better error reporting.

Reply:
Ah, I'm not used to having `setContinueAfterFailure: NO`

Reply:
Will do this as a follow-up.